### ISSUE TITLE
Change encoding method

### ISSUE THREAD

**Issue by jamesrwilliams** [jamesrwilliams/loquor#4]
I have this idea to abstract the response encoding method a little during processing as a way to allow support for applying multiple rulesets against a string at once.

Currently, the `parser` returns `[3,13]` numerical arrays as results by default. While the AS script works like this I'd like to introduce a middle format to use before the responses are sent. See below:

| Step | Input | Encoded |
| ----- | ----- |------- |
| Default to unsafe | `<p>Hello World</p>` | `000000000000000000`
| Parse layer 1         | `<p>Hello World</p>`  | `000111111111110000` | 

This output is then encoded into the same numerical array of `[3,13]` but the output encoded string can be run through multiple rule sets removing possible issues finding overlaps in the numerical arrays.

**Comment by jamesrwilliams** [comment]
Something along the lines of this:
```js

const tests = `<p>Hello world</p>`;
const input = `000111111111111000`;
const regex = /(1)\1+/g;

const matches = input.matchAll(regex);
const output = [];

for (const match of matches) {
	let start = match.index;
  let end = match.index + (match[0].length - 2);
	output.push([start, end]);
}

// Expected return [[3, 13]];
console.log(output);
```

**Comment by jamesrwilliams** [comment]
This is now being worked on via this branch: [`api-change`](https://github.com/jamesrwilliams/pts-loquor/tree/api-change)

**Comment by jamesrwilliams** [comment]
Closed with #5
